components/DownloadsComponent.tsx

```diff
@@ -0,0 +1,165 @@
+    ) => {
+        if (newAlignment) {
+        setArch(newAlignment);
+        }
+    };
+
+    const handleOs = (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        newOs: string | null,
+    ) => {
+    if (newOs !== os) {
+        setOs(newOs || '');
+        } else {
+        setOs('');
+        }
+    };
+
+    const handleVersion = (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        newVersion: string | null,
+    ) => {
+        if (newVersion !== version) {
+        setVersion(newVersion || '');
+        } else {
+        setVersion('');
+        }
+    }
+
+    useEffect(() => {
+        // Function to fetch release assets
+        let ignore = false;
+        const fetchAssets = async () => {
+            if (!ignore) {
+                setLoading(true);
+            try {
+                const assets = await allFilteredAssets();
+                setOriginalData(assets); // Save the original data
+                setDisplayData(assets);  // Start with all data displayed
+            } catch (e) {
+                setError('Failed to fetch release assets');
+                console.error(e);
+            } finally {
+                setLoading(false);
+            }
+            } 
+        };
+        // Call the fetch function
+        fetchAssets();
+
+        // Cleanup function to avoid setting state on unmounted component
+        return () => {
+            ignore = true;
+        };
+    }, []);
+
+    useEffect(() => {
+        const uniqueMinors = new Set<string>();
+        for (const asset of originalData) {
+        const versionComponents = asset.version.split('.');
+        if (versionComponents.length === 3) { // Ensure it's Major.Minor.Patch
+            const majorMinor = `${versionComponents[0]}.${versionComponents[1]}`;
+            uniqueMinors.add(majorMinor);
+        }
+        if (uniqueMinors.size === 4) break; // Stop after collecting 4 unique minors
+        }
+        setUniqueMinorVersions(Array.from(uniqueMinors));
+    }, [originalData]);
+
+    useEffect(() => {
+        const filtered = originalData.filter(asset => {
+            // Filter for architecture
+            const archFilter = arch && typeof arch === 'string' && arch.includes('ppc64')
+                ? asset.architecture.includes('ppc64el') || asset.architecture.includes('ppc64le')
+                : asset.architecture === arch;
+
+            // Filter for operating system, if os is set
+            const osFilter = os ? asset.operatingSystem.toLowerCase() === os : true;
+
+            // Filter for version, if version is set
+            const versionFilter = version ? asset.version.includes(version.toLowerCase()) : true;
+
+            // Return assets that match both filters
+            return archFilter && osFilter && versionFilter;
+        });
+
+        const displayData = arch === '' && os === '' && version === '' ? originalData : filtered;
+        setDisplayData(displayData);
```

Comment:
Use useMemo here so that you don't have to create a state element for a value that is calculated from the original data.

Reply:
Done